samples/nrf9160/gnss/src/assistance.c

```diff
@@ -0,0 +1,345 @@
+
+	if (!date_time_is_valid()) {
+		printk("Failed to get current time\n");
+		return -1;
+	}
+
+	struct nrf_cloud_pgps_init_param pgps_param = {
+		.event_handler = pgps_event_handler,
+		.storage_base = PM_MCUBOOT_SECONDARY_ADDRESS,
+		.storage_size = PM_MCUBOOT_SECONDARY_SIZE
+	};
+
+	if (nrf_cloud_pgps_init(&pgps_param) != 0) {
+		printk("Failed to initialize P-GPS\n");
+		return -1;
+	}
+#endif /* CONFIG_NRF_CLOUD_PGPS */
+
+	return 0;
+}
+
+int assistance_request(struct nrf_modem_gnss_agps_data_frame *agps_request)
+{
+	int err = 0;
+
+#if defined(CONFIG_NRF_CLOUD_PGPS)
+	/* Store the A-GPS data request for P-GPS use. */
+	memcpy(&agps_need, agps_request, sizeof(agps_need));
+
+#if defined(CONFIG_NRF_CLOUD_AGPS)
+	if (!agps_request->data_flags) {
+		/* No assistance needed from A-GPS, skip directly to P-GPS. */
+		nrf_cloud_pgps_notify_prediction();
+		return 0;
+	}
+
+	/* P-GPS will handle ephemerides, so skip those. */
+	agps_request->sv_mask_ephe = 0;
+	/* Almanacs are not needed with P-GPS, so skip those. */
+	agps_request->sv_mask_alm = 0;
+#endif /* CONFIG_NRF_CLOUD_AGPS */
+#endif /* CONFIG_NRF_CLOUD_PGPS */
+#if defined(CONFIG_NRF_CLOUD_AGPS)
+	int validity_time;
+
+	assistance_active = true;
+
+	validity_time = 5 * 60; /* 5 minutes */
+	err = nrf_cloud_jwt_generate(validity_time, jwt_buf, sizeof(jwt_buf));
+	if (err) {
+		printk("Failed to generate JWT, error: %d\n", err);
+		goto agps_exit;
+	}
+
+	struct nrf_cloud_rest_context rest_ctx = {
+		.connect_socket = -1,
+		.keep_alive = false,
+		.timeout_ms = CONFIG_NRF_CLOUD_REST_RECV_TIMEOUT * MSEC_PER_SEC,
+		.auth = jwt_buf,
+		.rx_buf = rx_buf,
+		.rx_buf_len = sizeof(rx_buf),
+		.fragment_size = 0, /* use CONFIG_NRF_CLOUD_REST_FRAGMENT_SIZE */
+		.status = 0,
+		.response = NULL,
+		.response_len = 0,
+		.total_response_len = 0
+	};
+
+	struct nrf_cloud_rest_agps_request request = {
+		.type = NRF_CLOUD_REST_AGPS_REQ_CUSTOM,
+		.agps_req = agps_request,
+		.net_info = NULL
+	};
+
+	struct nrf_cloud_rest_agps_result result = {
+		.buf = agps_data_buf,
+		.buf_sz = sizeof(agps_data_buf),
+		.agps_sz = 0
+	};
+
+	struct lte_lc_cells_info net_info = { 0 };
+
+	err = serving_cell_info_get(&net_info.current_cell);
+	if (err) {
+		printk("Could not get cell info, error: %d\n", err);
+	} else {
+		/* Network info for the location request. */
+		request.net_info = &net_info;
+	}
+
+	printk("Requesting A-GPS data, ephe 0x%08x, alm 0x%08x, flags 0x%02x\n",
+		agps_request->sv_mask_ephe,
+		agps_request->sv_mask_alm,
+		agps_request->data_flags);
+
+	err = nrf_cloud_rest_agps_data_get(&rest_ctx, &request, &result);
+	if (err) {
+		printk("Failed to get A-GPS data, error: %d\n", err);
+		goto agps_exit;
+	}
+
+	printk("Processing A-GPS data\n");
+
+	err = nrf_cloud_agps_process(result.buf, result.agps_sz);
+	if (err) {
+		printk("Failed to process A-GPS data, error: %d\n", err);
+		goto agps_exit;
+	}
+
+agps_exit:
+	assistance_active = false;
+#endif /* CONFIG_NRF_CLOUD_AGPS */
+
```

Comment:
I believe you need to call `nrf_cloud_rest_disconnect(&rest_ctx)`

Reply:
`rest_ctx.keep_alive` is set to `false`, so the connection is automatically disconnected by the nRF Cloud REST lib.